### ISSUE TITLE
Exporting a Backup does not work with 0.950.0

### ISSUE THREAD

**Issue by lefherz** [deltachat/deltachat-android#1114]
- Android version: LineageOS 14.1
- Device: gt-i9300
- Delta Chat version: 0.950.0

## Expected behavior

Exporting a backup saves a file with my chat logs, media, contacts, and autocrypt peer status.

## Actual behavior

Instead I get the following error: "IMEX FAILED to complete: \"could not copy file from \\\'/data/user/0/chat.delta/files/messenger.db\\\' to \\\'/storage/emulated/0/Download/delta-chat-2019-11-11-0.bak\\\'\""

## Steps to reproduce the problem:

- Go to Settings/Chat and Media
- Tap on "Backup"
- Enter Screen lock code to unlock the backup functionality
- read the info box on backups, and press "start backup"
- A loading screen appears. Directly after 1%, the following error appears: "IMEX FAILED to complete: \"could not copy file from \\\'/data/user/0/chat.delta/files/messenger.db\\\' to \\\'/storage/emulated/0/Download/delta-chat-2019-11-11-0.bak\\\'\""

- Screenshots:
- Logs:

```
device=samsung GT-I9300 (m0xx)
android=7.1.2 (xxxxx, lineage_i9300-userdebug 7.1.2 NJH47F 20190302 dev-keys)
sdk=25
memory=7M (13.75% free, 256M max)
memoryClass=192
host=xxxxx
applicationId=chat.delta
app=Delta Chat 0.950.0-gplay
installer=com.android.packageinstaller
ignoreBatteryOptimizations=true

[...]

11-11 16:17:51.626  2141  2165 I DeltaChat: INBOX-IDLE ended.
11-11 16:17:51.630  2141  2165 I DeltaChat: dc_perform_imap_jobs starting.
11-11 16:17:51.630  2141  2167 I DeltaChat: IMAP-IDLE has data.
11-11 16:17:51.630  2141  2167 I DeltaChat: SENTBOX-IDLE ended.
11-11 16:17:51.632  2141  2165 I DeltaChat: dc_perform_imap_jobs ended.
11-11 16:17:51.632  2141  2165 I DeltaChat: INBOX-fetch started...
11-11 16:17:51.633  2141  2167 I DeltaChat: dc_perform_sentbox_jobs EMPTY (for now).
11-11 16:17:51.635  2141  2167 I DeltaChat: SENTBOX-fetch started...
11-11 16:17:51.635  2141  2167 I DeltaChat: SENTBOX-fetch done in 0 ms.
11-11 16:17:51.639  2141  2167 I DeltaChat: SENTBOX-IDLE started...
11-11 16:17:51.640  2141  2167 I DeltaChat: IMAP-IDLE SPAWNING
11-11 16:17:51.674  2141  2165 I DeltaChat: 0 mails read from "INBOX".
11-11 16:17:51.675  2141  2165 I DeltaChat: INBOX-fetch done in 42.8902 ms.
11-11 16:17:51.677  2141  2165 I DeltaChat: INBOX-IDLE started...
11-11 16:17:51.677  2141  2165 I DeltaChat: IMAP-IDLE SPAWNING
11-11 16:17:51.938  2141  2166 I DeltaChat: IMAP-IDLE has data.
11-11 16:17:51.938  2141  2166 I DeltaChat: MVBOX-IDLE ended.
11-11 16:17:51.939  2141  2166 I DeltaChat: dc_perform_mbox_jobs EMPTY (for now).
11-11 16:17:51.957  2141  2166 I DeltaChat: MVBOX-fetch started...
11-11 16:17:51.991  2141  2166 I DeltaChat: 0 mails read from "DeltaChat".
11-11 16:17:51.991  2141  2166 I DeltaChat: MVBOX-fetch done in 34 ms.
11-11 16:17:51.993  2141  2166 I DeltaChat: MVBOX-IDLE started...
11-11 16:17:51.993  2141  2166 I DeltaChat: IMAP-IDLE SPAWNING
11-11 16:17:52.094  2141  2141 I DeltaChat: Interrupting INBOX-IDLE...
11-11 16:17:52.095  2141  2165 I DeltaChat: IMAP-IDLE interrupted
11-11 16:17:52.095  2141  2165 I DeltaChat: INBOX-IDLE ended.
11-11 16:17:52.096  2141  2165 I DeltaChat: dc_perform_imap_jobs starting.
11-11 16:17:52.100  2141  2165 I DeltaChat: INBOX-job #2753, action ImexImap started...
11-11 16:17:52.147  2141  2165 I DeltaChat: Suspending SENTBOX-thread.
11-11 16:17:52.147  2141  2165 I DeltaChat: Interrupting SENTBOX-IDLE...
11-11 16:17:52.147  2141  2167 I DeltaChat: IMAP-IDLE interrupted
11-11 16:17:52.147  2141  2167 I DeltaChat: SENTBOX-IDLE ended.
11-11 16:17:52.150  2141  2167 I DeltaChat: dc_perform_sentbox_jobs EMPTY (for now).
11-11 16:17:52.202  2141  2146 I art     : Do full code cache collection, code=107KB, data=125KB
11-11 16:17:52.212  2141  2146 I art     : Starting a blocking GC JitCodeCache
11-11 16:17:52.212  2141  2146 I art     : After code cache collection, code=95KB, data=92KB
11-11 16:17:52.436  2141  2146 I art     : Do partial code cache collection, code=110KB, data=114KB
11-11 16:17:52.437  2141  2146 I art     : After code cache collection, code=109KB, data=113KB
11-11 16:17:52.437  2141  2146 I art     : Increasing code cache capacity to 512KB
11-11 16:17:52.447  2141  2165 I DeltaChat: Suspending MVBOX-thread.
11-11 16:17:52.447  2141  2165 I DeltaChat: Interrupting MVBOX-IDLE...
11-11 16:17:52.447  2141  2166 I DeltaChat: IMAP-IDLE interrupted
11-11 16:17:52.447  2141  2166 I DeltaChat: MVBOX-IDLE ended.
11-11 16:17:52.449  2141  2166 I DeltaChat: dc_perform_mbox_jobs EMPTY (for now).
11-11 16:17:52.450  2141  2167 I DeltaChat: SENTBOX-IDLE will not be started as it was interrupted while not ideling.
11-11 16:17:52.455  2141  2167 I DeltaChat: dc_perform_sentbox_jobs EMPTY (for now).
11-11 16:17:52.459  2141  2167 I DeltaChat: dc_perform_sentbox_jobs EMPTY (for now).
11-11 16:17:52.747  2141  2165 I DeltaChat: Import/export process started.
11-11 16:17:52.750  2141  2166 I DeltaChat: MVBOX-IDLE will not be started as it was interrupted while not ideling.
11-11 16:17:52.750  2141  2166 I DeltaChat: dc_perform_mbox_jobs EMPTY (for now).
11-11 16:17:52.754  2141  2166 I DeltaChat: dc_perform_mbox_jobs EMPTY (for now).
11-11 16:17:52.767  2141  2165 I DeltaChat: Start housekeeping...
11-11 16:17:52.789  2141  2165 I DeltaChat: 69 files in use.
11-11 16:17:52.790  2141  2165 I DeltaChat: Housekeeping done.
11-11 16:17:53.217  2141  2165 I DeltaChat: Database closed.
11-11 16:17:53.217  2141  2165 I DeltaChat: Backup '/data/user/0/chat.delta/files/messenger.db' to '/storage/emulated/0/Download/delta-chat-2019-11-11-3.bak'.
11-11 16:17:53.219  2141  2165 E DeltaChat: Cannot copy "/data/user/0/chat.delta/files/messenger.db" to "/storage/emulated/0/Download/delta-chat-2019-11-11-3.bak": Operation not permitted (os error 1)
11-11 16:17:53.236  2141  2165 I DeltaChat: Opened "/data/user/0/chat.delta/files/messenger.db".
11-11 16:17:53.237  2141  2165 E DeltaChat: "IMEX FAILED to complete: \"could not copy file from \\\'/data/user/0/chat.delta/files/messenger.db\\\' to \\\'/storage/emulated/0/Download/delta-chat-2019-11-11-3.bak\\\'\""
11-11 16:17:53.237  2141  2165 I DeltaChat: Unsuspending SENTBOX-thread.
11-11 16:17:53.237  2141  2165 I DeltaChat: Unsuspending MVBOX-thread.
11-11 16:17:53.241  2141  2167 I DeltaChat: dc_perform_sentbox_jobs EMPTY (for now).
11-11 16:17:53.242  2141  2167 I DeltaChat: SENTBOX-fetch started...
11-11 16:17:53.242  2141  2167 I DeltaChat: SENTBOX-fetch done in 0 ms.
11-11 16:17:53.244  2141  2167 I DeltaChat: SENTBOX-IDLE started...
11-11 16:17:53.245  2141  2166 I DeltaChat: dc_perform_mbox_jobs EMPTY (for now).
11-11 16:17:53.246  2141  2166 I DeltaChat: MVBOX-fetch started...
11-11 16:17:53.246  2141  2165 I DeltaChat: dc_perform_imap_jobs ended.
11-11 16:17:53.247  2141  2165 I DeltaChat: INBOX-fetch started...
11-11 16:18:05.639  2141  2141 W PassphraseRequiredActionBarActivity: onPause()
11-11 16:18:05.978  2141  2928 D OpenGLRenderer: endAllActiveAnimators on 0x6fa16800 (RippleDrawable) with handle 0x6fde1070
```

**Comment by lefherz** [comment]
I also have a 0.510.1 from F-Droid running on the same device, where backups work without a problem.

**Comment by r10s** [comment]
hm, looks similar to https://github.com/deltachat/deltachat-android/issues/1084

**Comment by lefherz** [comment]
Ah, I only looked at the open issues 3)

**Comment by r10s** [comment]
sure, so it seems as if the closed #1084 does not fix all cases. but we have this issue open now, so no need to reopen #1084 i'd say.

**Comment by lefherz** [comment]
Was the fix released in 0.950.0? It was released 6 days ago, I think #796 was only merged into master 5 days ago.

**Comment by r10s** [comment]
> Was the fix released in 0.950.0? It was released 6 days ago, I think #796 was only merged into master 5 days ago.

you're right, i thought it was older :)

so, it looks like if this issue is already fixed :)

**Comment by lefherz** [comment]
Should I test it with the next release or just close the issue?

**Comment by r10s** [comment]
typically, we close issues once the fix is merged to master, we do not wait for closing until a release is out. so the issue tracker is a todo-for-devs, not a todo-for-testers :) but as changes pop up in the changelog, testers will get some hints about what to retry.

as this issue is likely the same as the already fixed one, i think we can close it. you re-open it if the problem is still there on the next update.
